tests/jni_api.rs

```diff
@@ -315,122 +315,140 @@ pub fn java_byte_array_from_slice() {
+                //auto_ptr.commit().unwrap();
+            }
+
+            // Confirm modification of original Java array
+            let mut res: [$jni_type; 3] = [0 as $jni_type; 3];
+            env.$get_array(java_array, 0, &mut res).unwrap();
+            assert_eq!(res[0] as i32, 2);
+            assert_eq!(res[1] as i32, 3);
+            assert_eq!(res[2] as i32, 4);
         }
-
-        // Commit would be necessary here, if there were no closure
-        //auto_ptr.commit().unwrap();
-    }
-
-    // Confirm modification of original Java array
-    let mut res: [i8; 3] = [0; 3];
-    env.get_byte_array_region(java_array, 0, &mut res).unwrap();
-    assert_eq!(res[0], 2);
-    assert_eq!(res[1], 3);
-    assert_eq!(res[2], 4);
-}
-
-#[test]
-pub fn get_long_array_elements() {
-    let env = attach_current_thread();
-
-    // Create original Java array
-    let buf: &[i64] = &[1, 2, 3];
-    let java_array = env
-        .new_long_array(3)
-        .expect("JNIEnv#new_long_array must create a java array with given size");
-
-    // Insert array elements
-    let _ = env.set_long_array_region(java_array, 0, buf);
-
-    // Use a scope to test Drop
-    {
-        // Get long array elements auto wrapper
-        let auto_ptr = env
-            .get_long_array_elements(java_array, ReleaseMode::CopyBack)
-            .unwrap();
-
-        // Check array size
-        assert_eq!(auto_ptr.size().unwrap(), 3);
-
-        // Check pointer access
-        let ptr = auto_ptr.as_ptr();
-        assert_eq!(unsafe { *ptr.offset(0) }, 1);
-        assert_eq!(unsafe { *ptr.offset(1) }, 2);
-        assert_eq!(unsafe { *ptr.offset(2) }, 3);
-
-        // Check pointer From access
-        let ptr: *mut jlong = std::convert::From::from(&auto_ptr);
-        assert_eq!(unsafe { *ptr.offset(0) }, 1);
-        assert_eq!(unsafe { *ptr.offset(1) }, 2);
-        assert_eq!(unsafe { *ptr.offset(2) }, 3);
-
-        // Check pointer into() access
-        let ptr: *mut jlong = (&auto_ptr).into();
-        assert_eq!(unsafe { *ptr.offset(0) }, 1);
-        assert_eq!(unsafe { *ptr.offset(1) }, 2);
-        assert_eq!(unsafe { *ptr.offset(2) }, 3);
-
-        // Modify
-        unsafe {
-            *ptr.offset(0) += 1;
-            *ptr.offset(1) += 1;
-            *ptr.offset(2) += 1;
-        }
-
-        // Commit would be necessary here, if there were no closure
-        //auto_ptr.commit().unwrap();
-    }
-
-    // Confirm modification of original Java array
-    let mut res: [i64; 3] = [0; 3];
-    env.get_long_array_region(java_array, 0, &mut res).unwrap();
-    assert_eq!(res[0], 2);
-    assert_eq!(res[1], 3);
-    assert_eq!(res[2], 4);
-}
+    };
+}
+
+test_get_array_elements!(
+    get_array_elements,
+    jint,
+    new_int_array,
+    get_int_array_region,
+    set_int_array_region
+);
+
+test_get_array_elements!(
+    get_int_array_elements,
+    jint,
+    new_int_array,
+    get_int_array_region,
+    set_int_array_region
+);
```

Comment:
Are these two any different?

Reply:
No, not really. It's there just to show that you can call the generic / parametric method if you want.

I can remove it.

Reply:
I see, I didn't notice. Let's add a comment? 
// Test a generic get_array_elements 
// Test type-specific array accessors